docs/content/documentation/systems/AuxKernels/tensor_mechanics/RankTwoScalarAux.md

```diff
@@ -1,10 +1,232 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
-
-# RankTwoScalarAux
+# Rank Two Scalar Aux
 !syntax description /AuxKernels/RankTwoScalarAux
 
+##Description
+The AuxKernel `RankTwoScalarAux` provides methods to calculate several different scalar quantities for a Rank-2 tensor, described below.
+In some types of calculations, the scalar quantity is calculated in a user-specified direction (by default set to the (0, 0, 1) axis); in other calculation types the user can specify the start and end points of a line along which the scalar quantity is calculated.
+
+If desired, `RankTwoScalarAux` can be restricted to calculate the scalar quantity data for a Rank-2 tensor at a single specified quadrature point per element. This option is generally used only for debugging purposes.
+
+## Axial Stress
+The scalar type `AxialStress` calculates the scalar value of a Rank-2 tensor, $T$, in the direction of the axis specified by the user.
+The user should give the starting point, $P^1$, and the end point, $P^2$ which define the axis.
+
+\begin{equation}
+\label{eq:axial_stress_scalar_type}
+s = \hat{a}_i T_{ij} \hat{a}_j \quad \text{ where } \quad \hat{a}_i = \frac{P^2_i - P^1_i}{\left| P^2_i - P^1_i \right|}
+\end{equation}
+where $\hat{a}$ is the normalized direction vector for the axis defined by the points $P^1$ and $P^2$.
+
+## Direction
+The scalar type `direction` calculates the scalar value of a Rank-2 tensor, $T$, in the direction selected by the user as shown by Eq \eqref{eq:direction_scalar_type}:
+\begin{equation}
+\label{eq:direction_scalar_type}
+s = D_i T_{ij} D_j
+\end{equation}
+where $D$ is the direction vector specified in the input file.
+
+### Example Input File Syntax
+!listing modules/tensor_mechanics/test/tests/elastic_patch/elastic_patch.i block=AuxKernels/direction
+
+and an AuxVariable is required to store the AuxKernel information. Note that the name of the AuxVariable is used as the argument for the `variable` input parameter in the `RankTwoScalarAux` block.
+!listing modules/tensor_mechanics/test/tests/elastic_patch/elastic_patch.i block=AuxVariables/direction
+
+## Effective Strain
+The scalar type `EffectiveStrain` calculates an effective scalar measure of a Rank-2 tensor, often strain ($\epsilon_{ij}$) according to Eq
+\eqref{eq:effective_strain_scalar_type}.
+\begin{equation}
+\label{eq:effective_strain_scalar_type}
+s = \sqrt{\frac{2}{3} \epsilon_{ij} \epsilon_{ij}}
+\end{equation}
```

Comment:
I'd add a highlighted note (one of the ones with bright colors so you notice it) here:

This should not be confused with the effective plastic strain or effective creep strain, which are computed as integrals over the history of the inelastic strain, and are computed as scalar material properties (named 'effective_plastic_strain' and 'effective_creep_strain') for applicable $J_2$ models. These are computed as:
\begin{equation}
s = \int_t\sqrt{\frac{2}{3} \dot{\epsilon}^p_{ij} \dot{\epsilon}^p_{ij}} \mathrm{d}t
\end{equation}
where $t$ is time and $\dot{\epsilon}^p_{ij}$ is the inelastic strain increment.